airflow/providers/yandex/secrets/secrets_manager.py

```diff
@@ -0,0 +1,257 @@
+        If set to None (null in JSON), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/variables"
+    :param config_prefix: Specifies the prefix of the secret to read to get Configurations.
+        If set to None (null in JSON), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/config"
+    :param sep: Specifies the separator used to concatenate secret_prefix and secret_id.
+        Default: "/"
+    """
+
+    def __init__(
+        self,
+        yc_oauth_token: str | None = None,
+        yc_sa_key_json: str | None = None,
+        yc_sa_key_json_path: str | None = None,
+        yc_connection_id: str | None = None,
+        folder_id: str = "",
+        connections_prefix: str | None = "airflow/connections",
+        variables_prefix: str | None = "airflow/variables",
+        config_prefix: str | None = "airflow/config",
+        sep: str = "/",
+    ):
+        super().__init__()
+
+        self.yc_oauth_token = yc_oauth_token
+        self.yc_sa_key_json = yc_sa_key_json
+        self.yc_sa_key_json_path = yc_sa_key_json_path
+        self.yc_connection_id = yc_connection_id or default_conn_name
+        self._use_connection = not any([yc_oauth_token, yc_sa_key_json, yc_sa_key_json_path])
+
+        self.folder_id = folder_id
+        self.connections_prefix = connections_prefix.rstrip(sep) if connections_prefix is not None else None
+        self.variables_prefix = variables_prefix.rstrip(sep) if variables_prefix is not None else None
+        self.config_prefix = config_prefix.rstrip(sep) if config_prefix is not None else None
+        self.sep = sep
+
+    def get_conn_value(self, conn_id: str) -> str | None:
+        """
+        Retrieve from Secrets Backend a string value representing the Connection object.
+
+        :param conn_id: Connection ID
+        :return: Connection Value
+        """
+        if self.connections_prefix is None:
+            return None
+
+        if self._use_connection:
+            if conn_id == self.yc_connection_id:
+                return None
+
+        return self._get_secret_value(self.connections_prefix, conn_id)
+
+    def get_variable(self, key: str) -> str | None:
+        """
+        Return value for Airflow Variable.
+
+        :param key: Variable Key
+        :return: Variable Value
+        """
+        if self.variables_prefix is None:
+            return None
+
+        return self._get_secret_value(self.variables_prefix, key)
+
+    def get_config(self, key: str) -> str | None:
+        """
+        Return value for Airflow Config Key.
+
+        :param key: Config Key
+        :return: Config Value
+        """
+        if self.config_prefix is None:
+            return None
+
+        return self._get_secret_value(self.config_prefix, key)
+
+    @property
+    def _client(self):
+        """Create a Yandex Cloud SDK client."""
+        if self._use_connection:
```

Comment:
Reason for this part if initialization to happen here and not in __init__ is confusing. Is Connection object not availabe when `LockboxSecretManager.__init__` is called? If that's the case, I'd add explicit comment to clarify intent of this lazy initialization.